### PR TITLE
Create Transfigurator image

### PR DESCRIPTION
**Pull request by chases2**
Create an image for pushing TestGrid/Prow configurations from other Prow instances to TestGrid.

After this image is made, Prow jobs across multiple instances can use the image directly to maintain their TestGrid config.

xref https://github.com/istio/test-infra/pull/2006#discussion_r339261941

### CODE REVIEW

File: testgrid/cmd/transfigure/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+* `JOB_CONFIG`: Prow's\* Job Config path
+* `TESTGRID_CONFIG`: TestGrid configuration directory or default file
+* `ORG_REPO`: The subdirectory in `/config/testgrids/...` to push to. Usually `<github_org>/<github_repository>`.
+
+\*"Prow" refers to your non-k8s Prow instance
+
+## Building
+The gcr.io/k8s-prow/transfigure image is built and published automatically by [`post-test-infra-push-prow`](https://github.com/kubernetes/test-infra/blob/9a939de10fa72af415eb1e628345b7d16c1f0be0/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L118-L143) with the rest of the Prow components.
+
+You can build the image locally with `bazel run //testgrid/cmd/transfigure` (note `bazel run` not `bazel build`). Publish to a remote repository after building with `docker push` or build and push all Prow images at once with [`prow/push.sh`](/prow/push.sh).
```
Comment (fejta) [inline testgrid/cmd/transfigure/README.md]: Technically `bazel build :image` is how you build the image and `bazel run :image` is how you load it into your local docker instance

https://github.com/bazelbuild/rules_docker#using-with-docker-locally

File: testgrid/cmd/transfigure/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Transfigure.sh
+
+Transfigure is an image that generates a YAML TestGrid configuration from a Prow configuration and pushes it to be used on [testgrid.k8s.io](https://testgrid.k8s.io/).
```
Comment (fejta) [inline testgrid/cmd/transfigure/README.md]: You define the destination of `[testgrid.k8s.io]` below, remove one of them?

File: testgrid/cmd/transfigure/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,28 @@
+load("@io_bazel_rules_docker//container:image.bzl", "container_image")
+
+container_image(
+    name = "image",
+    base = "@gcloud-go//image",
+    directory = "/",
+    entrypoint = ["/transfigure.sh"],
+    files = [
+        "transfigure.sh",
+        "//robots/pr-creator",
```
Comment (fejta) [inline testgrid/cmd/transfigure/BUILD.bazel]: This looks like a better pattern than making the image download and compile pr-creator with the `go run` command. Can we a) switch the script to do this and b) do this for configurator as well?

File: testgrid/cmd/transfigure/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+It is used specifically for Prow instances other than the k8s instance of Prow.
+
+## Requirements
+`transfigure.sh` takes, as an argument, a GitHub personal access token that has been granted the `repo` access scope.
```
[the remark this answers is not in the document]
Comment (chases2) [reply testgrid/cmd/transfigure/README.md]: Work-in-progress, see https://github.com/istio/test-infra/pull/2006/

Once we get this Prow job working in the wild, I intend to refactor this and Configurator's readme to just point to a prowjob

File: testgrid/cmd/transfigure/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,28 @@
+load("@io_bazel_rules_docker//container:image.bzl", "container_image")
+
+container_image(
+    name = "image",
+    base = "@gcloud-go//image",
```
[the remark this answers is not in the document]
Comment (chases2) [reply testgrid/cmd/transfigure/BUILD.bazel]: It also needs linux, and `@git_base//image` seems to choke on linux commands?
I left a comment describing what I need from the base image

File: testgrid/cmd/transfigure/README.md (hunk 1 above)
Comment (chases2) [reply testgrid/cmd/transfigure/README.md]: Reworded; what do you think?

File: testgrid/cmd/transfigure/transfigure.sh [hunk 6]
```diff
@@ -14,17 +14,15 @@ set -o pipefail
 main() {
   branch="transfigure-branch"
 
-  if [[ $# -lt 1 ]]; then
-      echo "Usage: $(basename "$0") </path/to/github/token> [prow_config] [prow-job_config] [testgrid-yaml] [subdir-path]" >&2
-      echo "Path to GitHub token required" >&2
-      exit 1
+  if [[ ! $# -eq 5 ]]; then
```
Comment (fejta) [inline testgrid/cmd/transfigure/transfigure.sh]: `[[ $# != 5 ]]` or `[ $# -ne 5]`

**Pull request merged**